src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -111,14 +111,21 @@ private async Task<bool> DoExecute(ProcessedArgs localSettings)
             { nameof(cache.UnchangedFilesPath), cache.UnchangedFilesPath },
             { SonarProperties.PullRequestCacheBasePath, cache.PullRequestCacheBasePath }
         };
-        AnalysisConfigGenerator.GenerateFile(
+        var config = AnalysisConfigGenerator.GenerateFile(
             localSettings,
             buildSettings,
             additionalSettings,
             argumentsAndRuleSets.ServerSettings,
             argumentsAndRuleSets.AnalyzersSettings,
             server.ServerVersion.ToString(),
             resolvedJavaExePath);
+
+        if (config.ScanAllAnalysis)
+        {
+            logger.LogUIWarning(Resources.WARN_UI_ScanAllAnalysisEnabled);
+        }
+
+        logger.WriteUIWarnings(buildSettings.SonarOutputDirectory); // Create the UI warnings file to be picked up the plugin
```

Comment:
Optional: 
I would remove this comment, as I don't find it adding much value.
You are already linking to the plugin code inside, so this seems a bit redundant to me.